pkg/api/types.go

```diff
@@ -311,3 +327,53 @@ func (p *PullPolicy) Set(v string) error {
 	}
 	return nil
 }
+
+// IsSafeFilename checks if the filename provided does not contain any invalid
+// characters
+func IsSafeFilename(name string) bool {
```

Comment:
It is there to prevent injection of malicious directory names. There are just few characters that can do damage so for that I think blacklisting them makes more sense.